src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -4000,6 +4000,22 @@ public Type lub(Type... ts) {
 
         int[] kinds = new int[ts.length];
 
+        boolean haveValues = false;
```

Comment:
If the goal is to fix the behavior of conditional expression typing, I think the fix should happen outside lub(), more specifically in Attr::condType

Reply:
Hmm. - to me it looks more natural to push this down to the service. But I am open to bubbling this up to the client. With your consent, I will combine this with the fix for JDK-8244513